HIP-1.md

```diff
@@ -0,0 +1,146 @@
+
+The ability for relayers to inject arbitrary data into the `ISM` presents a natural opportunity to move the merkle proof out of the `Inbox.process()` interface and into the `ISM`.
+In this model, the `Inbox` would be responsible only for replay protection.
+
+```
+interface IInterchainSecurityModule {
+    function moduleData() external view returns (bytes memory);
+
+    function accept(
+        bytes32 _root,
+        bytes32[32] calldata _proof,
+        bytes calldata _message,
+        bytes calldata _data
+    ) external returns (bool);
+}
+
+interface IInbox {
+    function process(bytes calldata _message) external;
+}
+```
+
+This approach has the advantage of being even more modular than the proposed solution, as the `Inbox` is unopinionated about how messages are sent.
+This allows applications to create `ISMs` that do not incorporate the incremental merkle tree.
+As a simple example, one could build an `ISM` that is a multisig on individual messages, as opposed to a merkle root.
+
+The drawback is that those `ISMs` which wish to incorporate the incremental merkle tree into their security model must do that explicitly.
+We imagine most applications to prefer the censorship resistance and economies of scale that using the incremental merkle tree allows for.
+Thus, we propose leaving proof verification in the `Inbox` so that all `ISM` developers get this "for free".
+
+#### Removal of `_index` from Inbox.process()
+
+In Hyperlane v1, the height of `_root` is passed to `Inbox.process()` as `_index`.
+This is necessary because v1 validators sign the `(_root, _index)` tuple.
+Including `_index` in the validator signature allows Hyperlane to support slashing conditions that it wouldn't otherwise be able to support.
+However, many of the `ISMs` we can envision for v2 would not make use of this, e.g.
+
+- In optimistic models, signature verification is done _before_ calling `ISM.accept()`
+- In zk-based native verification models, the signatures being verified are not specific to Hyperlane
+
+Rather than include a parameter that only some `ISMs` make use of, we propose that, when needed, this parameter be included within `_data`, just as any other piece of `ISM` specific off-chain data would be.
+
+### **Backwards Compatibility**
+
+For backwards compatibility, each `Inbox` should be configured with a default `ISM`.
+If `recipient.interchainSecurityModule()` reverts, the default `ISM` should be used.
+
+### **Security Considerations**
+
+We believe that modularity is the best security philosophy for interchain messaging, as it has the following properties:
+
+- Fault isolation: The impact of `ISM` failure is limited to those applications using the `ISM`.
+- Forwards compatibility: Sovereign consensus allows for security models to change as new and better solutions emerge (e.g. zkp-based light clients).
+- Customizability: Applications can tailor security models to their needs, rather than relying on a one-size-fits-all solution.
```

Comment:
we should make every attempt to mitigate "faulty configurations" which seem to be the cause of a majority of exploits, rather than safety faults